proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1984 @@
+
+    * Let `color` be the result of [converting] `$color` to `space`.
+
+* Otherwise:
+
+  * Let `space` be `origin-space`.
+
+  * Let `color` be the value of `$color`.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+    * If `alpha == none`, throw an error.
+
+    * Set `alpha` to the result of [scaling] `alpha` by `$alpha` with `max` 1.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `scale`, `factor` in `channel-args`:
+
+  * If `scale` is not the name of a [scalable] channel in `channels`, throw an
+    error.
+
+  * Let `channel` be the corresponding `channel` in `channels` with a name
+    matching `scale`.
+
+  * Let `max` be the upper boundary of `channel` in `space`.
+
+  * If `channel > max` or `channel < 0`, throw an error.
+
+  * Set the corresponding `channel` in `channels` to the result of [scaling]
+    `channel` by `factor` with `max`.
+
+* Set `channels` be the result of [normalizing] `channels` in `space`.
+
+* Let `new` be a color in color space `space`, with `channels` channels, and an
+  alpha of `alpha`.
+
+* Return the result of [converting] `new` into `origin-space`.
+
+[scalable]: #known-color-space
+[scaling]: #scaling-a-number
+
+### `color.complement()`
+
+```
+complement($color, $space: null)
+```
+
+This function is also available as a global function named `complement()`.
+
+* If `$color` is not a color in a [known color space], throw an error.
+
+* If `$space` is null:
+
+  * If `$color` is a legacy color, let `space` be the [known color space]
+    named `hsl`.
+
+  * Otherwise, throw an error.
+
+* Otherwise:
+
+  * Let `space` be the result of [looking up a known color space] named
+    `$space`.
+
+  * If `space` is not a [known color space] with a polar-angle hue channel,
+    throw an error.
+
+    > This currently allows `hsl`, `hwb`, `lch`, and `oklch`. We may decide to
+    > provide additional options in the future.
+
+  * Let `space` be the value of `$space`.
+
+* Return the result of calling
+  `color.adjust($color, $hue: 180deg, $space: space)`.
+
+### `color.invert()`
+
+```
+invert($color, $space: null)
+```
+
+This function is also available as a global function named `invert()`.
+
+* If `$color` is not a color, throw an error.
+
+* If `$space` is null:
+
+  * If `$color` is a legacy color, let `space` be `rgb`, and let `mix-space`
+    be null.
+
+    > This allows us to also enforce legacy behavior in the final weighted mix.
+
+  * Otherwise, throw an error.
+
+* Otherwise:
+
+  * Let `space` be the result of [looking up a known color space] named
+    `$space`.
+
+  * If `space` is not a [known color space], throw an error.
+
+  * Let `mix-space` be `space`.
+
+* Let `color` be the result of [converting] and [gamut mapping] `$color`
+  into the color space `space`.
```

Comment:
Do we need to gamut-map here? We can theoretically meaningfully invert `color(srgb 2 -0.5 0.3)` into `color(srgb -1 1.5 0.7)`, right?